pkg/cluster/spec/validate.go

```diff
@@ -659,9 +659,44 @@ func (s *Specification) validateTiSparkSpec() error {
 	return nil
 }
 
+func (s *Specification) validateTLSEnabled() error {
+	if !s.GlobalOptions.TLSEnabled {
+		return nil
+	}
+
+	// check for component with no tls support
+	compList := make([]Component, 0)
+	s.IterComponent(func(c Component) {
+		if len(c.Instances()) > 0 {
+			compList = append(compList, c)
+		}
+	})
+
+	for _, c := range compList {
+		switch c.Name() {
+		case ComponentPD,
+			ComponentTiDB,
+			ComponentTiKV,
+			ComponentPump,
+			ComponentDrainer,
+			ComponentCDC,
+			ComponentPrometheus,
+			ComponentAlertManager,
+			ComponentGrafana:
+		default:
```

Comment:
TiFlash supports it after v4.0.4

Reply:
I didn't find any document about how to enable TLS for TiFlash, I think we could add that in another PR. It won't be too difficult after this PR been merged.